horovod/common/util.py

```diff
@@ -251,3 +254,17 @@ def split_list(l, n):
     """
     d, r = divmod(len(l), n)
     return [l[i * d + min(i, r):(i + 1) * d + min(i + 1, r)] for i in range(n)]
+
+
+def check_installed_version(name, version, exception=None):
+    file_path = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)),\
+        os.pardir, "metadata.json"))
+    with open(file_path) as f:
+        installed_version = json.load(f)[name]
+        if not installed_version == version:
+            if exception is None:
+                warnings.warn(f'Framework {name} installed with version {installed_version} but found version {version}.\n\
```

Comment:
Let's move this into `common/exceptions.py` so we can reuse it as the exception message as well:

```
def get_version_mismatch_message(name, version, installed_version):
    return ...
```